### PR TITLE
NEO coin type and related stuff

### PR DESCRIPTION
**Pull request by ArtemGoryunov**
## Description

Part of integration #44. Coin type and related stuff were added.

## Testing instructions

Run unit tests.

## Types of changes

A new `PublicKeyType` was added - `PublicKeyType::nist256p1`.
Need to define proper way to handle `PublicKeyType.type`, `PublicKeyType.isValid` based on `nist256p1Size`.

## Checklist

- [x] Write unit tests. Put them in a subfolder of `tests/`.
- [x] Add relevant constants in `TWCoinType`, `TWP2SHPrefix`, `TWEthereymChainID`, `TWHRP`, etc., as necessary.
- [x] Return correct curve and purpose in `src/Coin.cpp`.
- [x] Implement address validation and derivation in `src/Coin.cpp`.
- [x] Implement coin configuration `src/include/TWCoinTypeConfiguration.cpp`.
- [x] Write interface header in `include/TrustWalletCore` and implement the interface in `src/interface`.
    - [x] Address interface (if necessary).
- [x] Validate generated code in Android an iOS projects.

### CODE REVIEW

**Comment by madcake** [comment]
Tests for android and swift will be in the next PR?

**Comment by ArtemGoryunov** [comment]
> Tests for android and swift will be in the next PR?

yes, I think integration tests will be with Signing in oncoming PR.

File: swift/Sources/Blockchains/BlockchainFactory.swift [hunk 1]
```diff
@@ -57,5 +57,8 @@ public func blockchain(coin: CoinType) -> Blockchain {
         return Stellar()
     case .aion:
         return Aion()
+    case .neo:
+        fatalError()
+//        return NEO()
```
Comment (alejandro-isaza) [inline swift/Sources/Blockchains/BlockchainFactory.swift]: Please fix

Comment (ArtemGoryunov) [reply swift/Sources/Blockchains/BlockchainFactory.swift]: done

File: src/interface/TWNEOAddress.cpp [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+}
+
+bool TWNEOAddressIsValidString(TWString *_Nonnull string) {
+    auto s = reinterpret_cast<const std::string*>(string);
+    return Address::isValid(*s);
+}
+
+struct TWNEOAddress *_Nullable TWNEOAddressCreateWithString(TWString *_Nonnull string) {
+    auto s = reinterpret_cast<const std::string*>(string);
+    const auto address = Address(*s);
```
Comment (alejandro-isaza) [inline src/interface/TWNEOAddress.cpp]: This can throw, let's catch the exception and return `nullptr`

File: src/interface/TWNEOAddress.cpp [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    return new TWNEOAddress{ std::move(address) };
+}
+
+struct TWNEOAddress *_Nullable TWNEOAddressCreateWithData(TWData *_Nonnull data) {
+    auto d = reinterpret_cast<const std::vector<uint8_t>*>(data);
+    return new TWNEOAddress{ Address(*d) };
```
Comment (alejandro-isaza) [inline src/interface/TWNEOAddress.cpp]: This can throw, let's catch the exception and return `nullptr`

File: src/PublicKey.h [hunk 4]
```diff
@@ -30,6 +31,9 @@ class PublicKey {
     /// The number of bytes in a secp256k1 extended public key.
     static const size_t secp256k1ExtendedSize = 65;
 
+    /// The number of bytes in a nist256p1 public key.
+    static const size_t nist256p1Size = 33;
```
Comment (ArtemGoryunov) [inline src/PublicKey.h]: @alejandro-isaza what is the proper way to handle for `nist256p1Size` here:
https://github.com/TrustWallet/wallet-core/blob/243c4d313db2dea6c9bab1d6ee240f62bde1d0b8/src/PublicKey.h#L41-L75

and here 
https://github.com/TrustWallet/wallet-core/blob/243c4d313db2dea6c9bab1d6ee240f62bde1d0b8/src/PublicKey.h#L60-L74

Comment (alejandro-isaza) [reply src/PublicKey.h]: nist and secp256 are pretty similar so we can may just have `ecdsaSize` for both.

File: src/interface/TWNEOAddress.cpp (hunk 3 above)
Comment (ArtemGoryunov) [reply src/interface/TWNEOAddress.cpp]: Done

File: src/interface/TWNEOAddress.cpp (hunk 2 above)
Comment (ArtemGoryunov) [reply src/interface/TWNEOAddress.cpp]: done

File: src/PublicKey.h (hunk 4 above)
Comment (ArtemGoryunov) [reply src/PublicKey.h]: 👌

**Pull request merged**